packages/template-retail-react-app/app/hooks/use-script.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useEffect, useState} from 'react'
+import PropTypes from 'prop-types'
+
+/**
+ * Custom hook to handle script loading
+ * @param {string} src - The source URL for the script
+ * @returns {Object} The script load status
+ */
+const useScript = (src) => {
+    const [scriptLoadStatus, setScriptLoadStatus] = useState({loaded: false, error: false})
+    // Effect to load and initialize the script
+    useEffect(() => {
+        if (!src) {
+            return
+        }
+
+        // Check if script already exists
+        const scriptAlreadyOnPage = document.querySelector(`script[src="${src}"]`)
+
+        if (!scriptAlreadyOnPage) {
+            const script = document.createElement('script')
+            script.src = src
+            script.async = true
+            script.setAttribute('data-status', 'loading')
```

Comment:
This data attribute never seems to change. Is that needed at all? Doesn't seem to serve any purpose.